### DISCUSSION TITLE
Duplicate detection not working

### DISCUSSION THREAD

**Discussion by nylocx** [vale46n1/immich_duplicate_finder#36] [Q&A]
I'm currently testing out immich and the immich duplicate finder and for me it currently does not seem to work, but I have no error in the logs.

After I entered the login and API key to my immich I get: 
Assets fetched successfully!
Processing asset 14/14

If I call <my_immich_instance>/api/assets I get 17 assets. I have added 3 jpegs the same image in different resolutions form 256 to 512 to 1024 (square image). 

This is a sample of one of the "not recognized" images.
```
{
    "id": "0590d4d0-3995-4000-821b-bce65440b1bc",
    "deviceAssetId": "web-512-512.jpg-1667808394826",
    "ownerId": "36741db1-b6b5-420b-8cdd-91b3fd2269b0",
    "deviceId": "WEB",
    "libraryId": "d45739db-9b4f-447e-b373-83922935cf37",
    "type": "IMAGE",
    "originalPath": "upload/library/admin/2022/2022-11-07/512-512.jpg",
    "originalFileName": "512-512.jpg",
    "resized": true,
    "thumbhash": "FAgKDwISuWdfl2iTeUeYd4iGhpBoCIoF",
    "fileCreatedAt": "2022-11-07T08:06:34.826Z",
    "fileModifiedAt": "2022-11-07T08:06:34.826Z",
    "localDateTime": "2022-11-07T08:06:34.826Z",
    "updatedAt": "2024-05-13T21:58:55.894Z",
    "isFavorite": false,
    "isArchived": false,
    "isTrashed": false,
    "duration": "0:00:00.00000",
    "exifInfo": {
      "make": null,
      "model": null,
      "exifImageWidth": 512,
      "exifImageHeight": 512,
      "fileSizeInByte": 47927,
      "orientation": null,
      "dateTimeOriginal": "2022-11-07T08:06:34.826Z",
      "modifyDate": "2022-11-07T08:06:34.826Z",
      "timeZone": null,
      "lensModel": null,
      "fNumber": null,
      "focalLength": null,
      "iso": null,
      "exposureTime": null,
      "latitude": null,
      "longitude": null,
      "city": null,
      "state": null,
      "country": null,
      "description": "",
      "projectionType": null
    },
    "livePhotoVideoId": null,
    "tags": [],
    "people": [],
    "checksum": "d5EEiCxItVGMuzU64LFbDnjpsuE=",
    "stackCount": null,
    "isExternal": false,
    "isOffline": false,
    "isReadOnly": false,
    "hasMetadata": true
  }
  ```

Anyone an idea what is happening here? I would have expected that the images show up as clear duplicates.

**Comment by vale46n1** [comment]
Did you rebuild the vector Db?
Inviato da iPad

Il giorno 17 mag 2024, alle ore 20:51, Alexander Görtz ***@***.***> ha scritto:

﻿

I'm currently testing out immich and the immich duplicate finder and for me it currently does not seem to work, but I have no error in the logs.

After I entered the login and API key to my immich I get:
Assets fetched successfully!
Processing asset 14/14

If I call <my_immich_instance>/api/assets I get 17 assets. I have added 3 jpegs the same image in different resolutions form 256 to 512 to 1024 (square image).

This is a sample of one of the "not recognized" images.

{
    "id": "0590d4d0-3995-4000-821b-bce65440b1bc",
    "deviceAssetId": "web-512-512.jpg-1667808394826",
    "ownerId": "36741db1-b6b5-420b-8cdd-91b3fd2269b0",
    "deviceId": "WEB",
    "libraryId": "d45739db-9b4f-447e-b373-83922935cf37",
    "type": "IMAGE",
    "originalPath": "upload/library/admin/2022/2022-11-07/512-512.jpg",
    "originalFileName": "512-512.jpg",
    "resized": true,
    "thumbhash": "FAgKDwISuWdfl2iTeUeYd4iGhpBoCIoF",
    "fileCreatedAt": "2022-11-07T08:06:34.826Z",
    "fileModifiedAt": "2022-11-07T08:06:34.826Z",
    "localDateTime": "2022-11-07T08:06:34.826Z",
    "updatedAt": "2024-05-13T21:58:55.894Z",
    "isFavorite": false,
    "isArchived": false,
    "isTrashed": false,
    "duration": "0:00:00.00000",
    "exifInfo": {
      "make": null,
      "model": null,
      "exifImageWidth": 512,
      "exifImageHeight": 512,
      "fileSizeInByte": 47927,
      "orientation": null,
      "dateTimeOriginal": "2022-11-07T08:06:34.826Z",
      "modifyDate": "2022-11-07T08:06:34.826Z",
      "timeZone": null,
      "lensModel": null,
      "fNumber": null,
      "focalLength": null,
      "iso": null,
      "exposureTime": null,
      "latitude": null,
      "longitude": null,
      "city": null,
      "state": null,
      "country": null,
      "description": "",
      "projectionType": null
    },
    "livePhotoVideoId": null,
    "tags": [],
    "people": [],
    "checksum": "d5EEiCxItVGMuzU64LFbDnjpsuE=",
    "stackCount": null,
    "isExternal": false,
    "isOffline": false,
    "isReadOnly": false,
    "hasMetadata": true
  }


Anyone an idea what is happening here? I would have expected that the images show up as clear duplicates.

**Reply by nylocx** [reply]
Yes, I also tried to build the vectors by hand by executing the resnet model. And the diff between different scales was 8. something. Very strange.
